### PR TITLE
feat: enhance async validation logs

### PR DESCRIPTION
**Pull request by anchenyi**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27231503

### CODE REVIEW

**Review by supkasar — APPROVED** [review]

File: packages/fx-core/src/component/configManager/constant.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export enum SummaryConstant {
   Succeeded = "(√) Done:",
   Failed = "(×) Error:",
   NotExecuted = "(!) Warning:",
+  Warning = "(!) Warning:",
```
Comment (nliu-ms) [inline packages/fx-core/src/component/configManager/constant.ts:12]: It looks like the same as NotExecuted?

Comment (anchenyi) [reply packages/fx-core/src/component/configManager/constant.ts:12]: "NotExecuted" is used elsewhere, and the term "NotExecuted" does not precisely match our validation warning scenario. Therefore, I have added a new enum called "Warning" with the same string value as "NotExecuted".

**Comment by anchenyi** [comment]
New Output UI:
async_validation1

**Comment by anchenyi** [comment]
Mocked some validation results with all types.
async_validation2

**Review by nliu-ms — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by supkasar — APPROVED** [review]

**Pull request merged**